hack/verify-release.sh

```diff
@@ -0,0 +1,489 @@
+
+set -eu
+set -o pipefail
+
+# enable support for **/go.mod
+shopt -s globstar
+
+# user input
+VERSION="${1:?release version missing, provide without leading v. Example: 1.5.0}"
+GITHUB_TOKEN="${GITHUB_TOKEN:?export GITHUB_TOKEN with permissions to read unpublished release notes}"
+CONTAINER_RUNTIME="${CONTAINER_RUNTIME:-docker}"
+# correct remote will be autodetected, if empty
+REMOTE="${REMOTE:-}"
+
+# this repo
+ORG="metal3-io"
+PROJECT="${ORG}/cluster-api-provider-metal3"
+REGISTRY="quay.io"
+
+# if the given tag doesn't exist, we run only pre-tag checks
+TAG_EXISTS=""
+
+
+#
+# checklist
+#
+
+# required tools
+declare -a required_tools=(
+    awk
+    curl
+    git
+    jq
+    sed
+    "${CONTAINER_RUNTIME}"
+)
+
+# git tags
+declare -a git_annotated_tags=(
+    "v${VERSION}"
+)
+
+declare -a git_lightweight_tags=(
+    "api/v${VERSION}"
+    "test/v${VERSION}"
+)
+
+declare -a git_nonexisting_tags=(
+    "hack/tools/v${VERSION}"
+)
+
+# release notes should have these strings
+declare -a release_note_strings=(
+    ":recycle:"
+    "Changes since v"
+)
+
+# required strings that are postfixed with correct release number
+declare -a release_note_tag_strings=(
+    "The image for this release is: v${VERSION}"
+    "Ironic image tag is capm3-v${VERSION}"
+    "Mariadb image tag is capm3-v${VERSION}"
+)
+
+# release artefacts
+declare -a release_artefacts=(
+    cluster-template.yaml
+    example_variables.rc
+    infrastructure-components.yaml
+    metadata.yaml
+)
+
+# quay images
+declare -a container_images=(
+    "${ORG}/cluster-api-provider-metal3:v${VERSION}"
+    "${ORG}/ironic:capm3-v${VERSION}"
+    "${ORG}/mariadb:capm3-v${VERSION}"
+)
+
+# go mod bump checks - must match up to leading space before v
+declare -A module_groups=(
+    [capi]="
+        sigs.k8s.io/cluster-api
+        sigs.k8s.io/cluster-api/test
+    "
+    [k8s]="
+        k8s.io/api
+        k8s.io/apiextensions-apiserver
+        k8s.io/apimachinery
+        k8s.io/client-go
+        k8s.io/component-base
+    "
+)
+
+
+#
+# temporary files and cleanup trap
+#
+cleanup()
+{
+    rm -rf "${TMP_DIR}"
+}
+
+TMP_DIR="$(mktemp --tmpdir -d verify-release-XXXXX)"
+RELEASE_JSON="${TMP_DIR}/release.json"
+SCAN_LOG="${TMP_DIR}/scan.log"
+TAG_LOG="${TMP_DIR}/tag.log"
+mkdir -p "${TMP_DIR}"
+trap cleanup EXIT
+
+
+#
+# pre-requisites
+#
+check_tools()
+{
+    echo "Checking required tools ..."
+
+    for tool in "${required_tools[@]}"; do
+        type "${tool}" &>/dev/null || { echo "FATAL: need ${tool} to be installed"; exit 1; }
+    done
+
+    echo -e "Done\n"
+}
+
+detect_remote()
+{
+    # we support origin (default) and upstrea (if cloned with "gh" CLI tool)
+    echo "Detecting remote ..."
+
+    if [[ -z "${REMOTE}" ]]; then
+        REMOTE="$(git remote -v | grep "${PROJECT} (fetch)" | awk '{print $1;}')"
+
+        if ! [[ "${REMOTE}" =~ ^(origin|upstream)$ ]]; then
+            echo "WARNING: detected remote '${REMOTE}' is not supported"
+        fi
```

Comment:
The script failed for me here. I think the issue is that my remotes have `.git` at the end of the URL (exactly what you get if you copy the ssh URL from GitHub).
To solve this I suggest making the `grep` a bit more flexible, perhaps with a regex.

Somehow it just exited the script for me at this point. I do not get why it didn't echo the warning about unsupported remote :thinking:

Reply:
Fixed